src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidUninstantiatedInternalClasses.cs

```diff
@@ -204,7 +238,8 @@ static IEnumerable<INamedTypeSymbol> GetAllNamedTypeConstraints(ITypeParameterSy
                     var uninstantiatedInternalTypes = internalTypes
                         .Select(it => it.Key.OriginalDefinition)
                         .Except(instantiatedTypes.Select(it => it.Key.OriginalDefinition))
-                        .Where(type => !HasInstantiatedNestedType(type, instantiatedTypes.Keys));
+                        .Where(type => !HasInstantiatedNestedType(type, instantiatedTypes.Keys))
+                        .Where(type => !instantiatedTypeNames.ContainsKey(type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat).Substring("global::".Length)));
```

Comment:
This doesn't look correct. We shouldn't be relying on fully qualified names and should not be invoking ToDisplayString.